cw/capture.py

```diff
@@ -1167,118 +1169,269 @@ def capture_otbn_vertical(ot, ktp, fw_bin, pll_frequency, capture_cfg, device_cf
+            # Send the mask and start the keygen operation.
+            ot.target.simpleserial_write('k', mask)
+
+            # Wait until operation is done.
+            ret = ot.scope.capture(poll_done=True)
+            if ret:
+                raise RuntimeError('Timeout during capture')
+
+            # Check the number of cycles where the trigger signal was high.
+            cycles = ot.scope.adc.trig_count
+            tqdm.write("Observed number of cycles: %d" % cycles)
+
+            waves = ot.scope.get_last_trace(as_int=True)
+            # Read the output, unmask the key, and check if it matches
+            # expectations.
+            share0 = ot.target.simpleserial_read("r", seed_bytes, ack=False)
+            share1 = ot.target.simpleserial_read("r", seed_bytes, ack=False)
+            if share0 is None:
+                raise RuntimeError('Random share0 is none')
+            if share1 is None:
+                raise RuntimeError('Random share1 is none')
+
+            d0 = int.from_bytes(share0, byteorder='little')
+            d1 = int.from_bytes(share1, byteorder='little')
+            actual_key = (d0 + d1) % curve_order_n
+
+            tqdm.write(f'share0 = {share0.hex()}')
+            tqdm.write(f'share1 = {share1.hex()}')
+
+            if actual_key != expected_key:
+                raise RuntimeError('Bad generated key:\n'
+                                   f'Expected: {hex(expected_key)}\n'
+                                   f'Actual:   {hex(actual_key)}')
+
+            # Create a chipwhisperer trace object and save it to the project
+            # Args/fields of Trace object: waves, textin, textout, key
+            textout = share0 + share1  # concatenate bytearrays
+            trace = Trace(waves, mask, textout, seed_used)
+            check_range(waves, ot.scope.adc.bits_per_sample)
+            project.traces.append(trace, dtype=np.uint16)
+    elif capture_cfg["app"] == 'modinv':
+        # Check the lengths in the key/plaintext generator. In this case, "key"
+        # is the input to the modinv app. We only use the key part of the ktp
+        # to generate the key share inputs to the modinv app.
+        if ktp.keyLen() != modinv_share_bytes:
+            raise ValueError(
+                f'Unexpected input (share) length: {ktp.keyLen()}.\n'
+                f'Hint: set key len={modinv_share_bytes} in the configuration file.'
+            )
+
+        # select the otbn app on the device (0 -> keygen, 1 -> modinv)
+        ot.target.simpleserial_write("a", bytearray([0x01]))
+        time.sleep(0.3)
+
+        # set PRNG seed to ensure same sequence of randoms for split up measurements
+        random.seed(capture_cfg["batch_prng_seed"])
+
+        # set ecc256 fixed key and share inputs
+        # (uncomment the desired fixed shares depending on whether
+        # you want a random fixed key or a hardcoded fixed key)
+        # k_fixed_barray = ktp.next_key()
+        k_fixed_barray = bytearray(
```

Comment:
Since we already seed the PRNG to generate the same sequence every time we run the capture script, can't we just use above line and delete this?